promql/engine.go

```diff
@@ -374,6 +374,16 @@ func (ng *Engine) execEvalStmt(ctx context.Context, query *query, s *EvalStmt) (
 
 		evalTimer.Stop()
 		queryInnerEval.Observe(evalTimer.ElapsedTime().Seconds())
+		// Point might have a different timestamp, force it to the evaluation
+		// timestamp as that is when we ran the evaluation.
+		switch v := val.(type) {
+		case Scalar:
+			v.T = timeMilliseconds(s.Start)
+		case Vector:
+			for i := range v {
+				v[i].Point.T = timeMilliseconds(s.Start)
```

Comment:
The overhead should be small, but any reason not to do the conversation outside of the loop?